lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/cache/CachingFileSystemConfig.java

```diff
@@ -40,6 +43,45 @@ public CachingFileSystemConfig setCacheEnabled(boolean value)
         return this;
     }
 
+    public boolean isFileStatusCacheEnabled()
+    {
+        return fileStatusCacheEnabled;
+    }
+
+    @Config("cache.filestatus.enabled")
+    @ConfigDescription("Cache file status (size & last updated) to workers' local storage to fully cache immutable files")
```

Comment:
Why would we need a file status cache on the workers ?
The engine already knows file modified time and size from splits generation in the coordinator and sends that to the workers through ConnectorSplit implementations.

Reply:
Oh, I didn't realise that was the case! Thank you for the heads up, I'm reworking the PR to avoid this cache and retrieve the lastModifiedTime via the connectorSplit.

Reply:
I've removed the filestatus cache, this PR now uses the modified time from the connectorSplit 👍

Reply:
It might not be necessary to include the modifications to `lastModifiedTime`, and we might actually get better caching without it. We only set the `lastModifiedTime` for data files, but all of these are immutable, so knowing the file name suffices. Also, when writing in the Delta connector Trino will immediately [read back the footer of the Parquet file to collect statistics](https://github.com/trinodb/trino/blob/fa081358b45997bb4758e26d0c2a49f6be543aa8/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java#L202). This puts the footer in the cache. However, as the `lastModifiedTime` of the recently written Parquet file is typically unknown when reading statistics, Trino will use the `lastModifiedTime` metadata from S3. This `lastModifiedTime` could be different from the `lastModifiedTime` we store in the delta log. So the next time we attempt to read this file from the table we might get a cache miss, even if the data is cached.

There are two approaches to take here:
* Ensure that all `TrinoInputFile`s have `lastModifiedTime` set in a consistent manner. This is a bit tricky, as there are multiple code paths which make use of `readStatistics` (when doing `MERGE INTO`, deletion vectors).
* Leave it to the connector to specify the cache identifier to use (e.g., we could default to use `lastModifiedTime` when deriving the cache identifier, but the connector would override this for data files).